src/xtgeo/grid3d/grid_properties.py

```diff
@@ -1,24 +1,106 @@
 # -*- coding: utf-8 -*-
 
 """Module for Grid Properties."""
-
-
 import hashlib
 import warnings
-from typing import Optional
+from typing import List, Optional
+
+import deprecation
+import numpy as np
+import pandas as pd
 
 import xtgeo
 from xtgeo.common import XTGDescription, XTGeoDialog
 
 from . import _grid3d_utils as utils
-from . import _grid_etc1, _gridprops_etc, _gridprops_import_eclrun
+from . import _grid_etc1, _gridprops_import_eclrun
 from ._grid3d import _Grid3D
 from .grid_property import GridProperty
 
 xtg = XTGeoDialog()
 logger = xtg.functionlogger(__name__)
 
 
+def gridproperties_from_file(
+    pfile,
+    grid,
+    fformat=None,
+    names=None,
+    dates=None,
+    namestyle=0,
+    strict=(True, False),
+):
+    """Import grid properties from file.
+
+    In case of names='all' then all vectors which have a valid length
+    (number of total or active cells in the grid) will be read
+
+    Args:
```

Comment:
Nitpick: maintain order of params / doc of params

Reply:
I see I made a mistake trying to introduce grid as a required parameter here. I think it would be best to keep as is, so changed it back.